Source/Diffusion/ComputeStress_N.cpp

```diff
@@ -20,32 +21,60 @@ using namespace amrex;
  */
 void
 ComputeStressConsVisc_N (Box bxcc, Box tbxxy, Box tbxxz, Box tbxyz, Real mu_eff,
+                         const Array4<const Real>& cell_data,
                          Array4<Real>& tau11, Array4<Real>& tau22, Array4<Real>& tau33,
                          Array4<Real>& tau12, Array4<Real>& tau13, Array4<Real>& tau23,
                          const Array4<const Real>& er_arr)
 {
     Real OneThird   = (1./3.);
 
-    // Cell centered strains
-    ParallelFor(bxcc, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+    if (cell_data)
+    // constant alpha (stored in mu_eff)
     {
-        tau11(i,j,k) = -mu_eff * ( tau11(i,j,k) - OneThird*er_arr(i,j,k) );
-        tau22(i,j,k) = -mu_eff * ( tau22(i,j,k) - OneThird*er_arr(i,j,k) );
-        tau33(i,j,k) = -mu_eff * ( tau33(i,j,k) - OneThird*er_arr(i,j,k) );
-    });
+        // Cell centered strains
+        ParallelFor(bxcc, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+        {
+            Real rhoAlpha  = cell_data(i, j, k, Rho_comp) * mu_eff;
+            tau11(i,j,k) = -rhoAlpha * ( tau11(i,j,k) - OneThird*er_arr(i,j,k) );
+            tau22(i,j,k) = -rhoAlpha * ( tau22(i,j,k) - OneThird*er_arr(i,j,k) );
+            tau33(i,j,k) = -rhoAlpha * ( tau33(i,j,k) - OneThird*er_arr(i,j,k) );
+        });
 
-    // Off-diagonal strains
-    ParallelFor(tbxxy,tbxxz,tbxyz,
-    [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept {
-        tau12(i,j,k) *= -mu_eff;
-    },
-    [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept {
-        tau13(i,j,k) *= -mu_eff;
-    },
-    [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept {
-        tau23(i,j,k) *= -mu_eff;
-    });
+        // Off-diagonal strains
+        ParallelFor(tbxxy,tbxxz,tbxyz,
+        [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept {
+            tau12(i,j,k) *= -cell_data(i, j, k, Rho_comp) * mu_eff;
+        },
+        [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept {
+            tau13(i,j,k) *= -cell_data(i, j, k, Rho_comp) * mu_eff;
+        },
+        [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept {
+            tau23(i,j,k) *= -cell_data(i, j, k, Rho_comp) * mu_eff;
+        });
```

Comment:
For the off-diagonal stresses, shouldn't the **cell_data** be averaged to the location of the strain (this would hold for other places in the code as well)?
![image](https://github.com/erf-model/ERF/assets/103702284/78fce89e-c942-42ea-bd32-f13f16c6c691)

Reply:
Yes, you're right. I have a fix coming.